check.go

```diff
@@ -1,32 +1,67 @@
 package filenameslinter
 
 import (
+	"errors"
 	"fmt"
+	"io/fs"
 	"path"
-    "io/fs"
-    "regexp"
+	"regexp"
+	"sort"
 
 	"github.com/csunibo/synta"
 	syntaRegexp "github.com/csunibo/synta/regexp"
 )
 
-func CheckDir(synta synta.Synta, fs fs.ReadDirFS, dirPath string) (err error) {
-    entries, err := fs.ReadDir(dirPath)
+func CustomReadDir(fsys fs.FS, name string) ([]fs.DirEntry, error) {
+	if fsys, ok := fsys.(fs.ReadDirFS); ok {
+		return fsys.ReadDir(name)
+	}
+
+	file, err := fsys.Open(name)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	dir, ok := file.(fs.ReadDirFile)
+	if !ok {
+		return nil, &fs.PathError{Op: "readdir", Path: name, Err: errors.New("not implemented")}
+	}
+
+	list, err := dir.ReadDir(-1)
+	sort.Slice(list, func(i, j int) bool { return list[i].Name() < list[j].Name() })
```

Comment:
La `readDir` nativa quindi fa sort in automatico? qua la emuliamo o cosa?

Reply:
Se la speranza era di controllare i file in ordine, questo non funziona sempre, perchè nel primo if ritorniamo brutalmente e questo sort non verrebbe applicato.
In ogni caso non vedo il punto di organizzare, non cambia veramente i nostri controlli, però se è il comportamento di base della `readDir` allora lascerei.

Reply:
È il comportamento base di `readDiir`, ho copiato il sorgente proprio, lascerei così